Helics compatible tools (#1019)
GMLC-TDC/HELICS: docs/Tools_using_HELICS.md

-   [Cyme](http://www.cyme.com/software/cymdist/) Cyme has been used in connection with a python wrapper interface and through FMI wrapper.  

### Electric Transmission System Simulation
-   [GridDyn](https://github.com/LLNL/GridDyn), an open-source transmission power flow and dynamics simulator. HELICS support currently (4/14/2018) provided through the [`cmake_updates` branch](https://github.com/LLNL/GridDyn/tree/cmake_update).
-   [PSST](https://github.com/kdheepak/psst), an open-source python-based unit-commitment and dispatch market simulator. HELICS examples are included in the  [HELICS-Tutorial](https://github.com/GMLC-TDC/HELICS-Tutorial)
-   [MATPOWER](http://www.pserc.cornell.edu/matpower/), an open-source Matlab based power flow and optimal power flow tool. HELICS support currently (4/14/2018) under development.
-   [InterPSS](http://www.interpss.org/), a Java-based power systems simulator. HELICS support currently (4/14/2018) under development.  [Use case instructions](https://gmlc-tdc.github.io/HELICS-Use-Cases/PNNL-TD-Dynamic-Load/index.html).

> -   [PSLF](https://github.com/GMLC-TDC/PSLF-wrapper) Some Level of PSLF support is available using the experimental python interface.

Reviewer: ```suggestion
-   [PSLF](https://github.com/GMLC-TDC/PSLF-wrapper) has some level of support using the experimental python interface.
```

-   [PSS/e](https://new.siemens.com/global/en/products/energy/services/transmission-distribution-smart-grid/consulting-and-planning/pss-software/pss-e.html)
-   [Power World](https://www.powerworld.com/) PowerWorld Simulator is an interactive power system simulation package designed to simulate high voltage power system operation on a time frame ranging from several minutes to several days.

> -   [PyPower](https://pypi.org/project/PYPOWER/) There is no standard HELICS integration but it has been used on various projects. *<b>NOTE:</b> according to the authors of PyPower is no longer being actively maintained.*   PYPOWER is a power flow and Optimal Power Flow (OPF) solver. It is a port of MATPOWER to the Python programming language. Current features include:

Reviewer: ```suggestion
-   [PyPower](https://pypi.org/project/PYPOWER/) does not have a standard HELICS integration but it has been used on various projects. *<b>NOTE:</b> according to the authors of PyPower is no longer being actively maintained.*   PYPOWER is a power flow and Optimal Power Flow (OPF) solver. It is a port of MATPOWER to the Python programming language. Current features include:
```

    -   DC and AC (Newton’s method & Fast Decoupled) power flow and
    -   DC and AC optimal power flow (OPF)

### Electric Power Market simulation

> -   [FESTIV](https://github.com/NREL/FESTIV_MODEL) The Flexible Energy Scheduling Tool for Integrating Variable generation provides multi-timescale steady-state power system operations simulations that aims to replicate the full time spectrum of scheduling and reserve processes (multi-step commitment and dispatch plus simplified AGC) to meet energy and reliability needs of the bulk power system

Reviewer: ```suggestion
-   [FESTIV](https://github.com/NREL/FESTIV_MODEL), the Flexible Energy Scheduling Tool for Integrating Variable Generation, provides multi-timescale steady-state power system operations simulations that aims to replicate the full time spectrum of scheduling and reserve processes (multi-step commitment and dispatch plus simplified AGC) to meet energy and reliability needs of the bulk power system.
```

-   [Plexos](https://energyexemplar.com/solutions/plexos/) A commercial production cost simulator. Support via OpenPlexos is under development
-   [MATPOWER](http://www.pserc.cornell.edu/matpower/) (described above) also includes basic optimal powerflow support.
-   [PyPower](https://pypi.org/project/PYPOWER/)  (described above) also includes basic AC and DC optimal powerflow solvers.  

### Contingency Analysis tools
-   [CAPE](https://new.siemens.com/global/en/products/energy/services/transmission-distribution-smart-grid/consulting-and-planning/pss-software/psscape.html) protection system modeling (IN development)
-   [DCAT](https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-26197.pdf) Dynamic contingency analysis tool

## Communication Tools

-   HELICS provides built-in support for simple communications manipulations such as delays, lossy channels, etc. through its built-in filters.
-   [ns-3](https://www.nsnam.org/), a discrete-event communication network simulator. Supported via the [HELICS ns-3 module](https://github.com/GMLC-TDC/helics-ns3)

> -   [OMNet++](https://omnetpp.org/)  OMNeT++ is a public-source, component-based, modular and open-architecture simulation environment with strong GUI support and an embeddable simulation kernel. Its primary application area is the simulation of communication networks, but it has been successfully used in other areas like the simulation of IT systems, queueing networks, hardware architectures and business processes as well.

Reviewer: ```suggestion
-   [OMNet++](https://omnetpp.org/)  is a public-source, component-based, modular and open-architecture simulation environment with strong GUI support and an embeddable simulation kernel. Its primary application area is the simulation of communication networks, but it has been successfully used in other areas like the simulation of IT systems, queueing networks, hardware architectures and business processes as well.
```

  Early stage development with OMNET++ and HELICS is underway and a prototype example is available in [HELICS-omnetpp](https://github.com/GMLC-TDC/helics-omnetpp)

## Gas Pipeline Modeling

-   [NGFAST](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.172.1169)
-   [GasModels.jl](https://github.com/lanl-ansi/GasModels.jl)

## Optimization packages

-   [GAMS](https://www.gams.com/)
-   [JuMP](https://www.juliaopt.org/) support is provided through the HELICS Julia interface

## Transportation modeling
These are in development and are not operational yet

> -   [Beam](http://beam.lbl.gov/)  An upcoming project will be developing a beam interface to HELICS.

Reviewer: ```suggestion
-   [BEAM](http://beam.lbl.gov/) will have an interface to HELICS developed in an upcoming project.
```

> -   [Polaris](https://www.anl.gov/es/polaris-transportation-system-simulation-tool) An upcoming project will be developing a Polaris interface to HELICS.

Reviewer: ```suggestion
-   [POLARIS](https://www.anl.gov/es/polaris-transportation-system-simulation-tool) will have an interface to HELICS in an upcoming project.
```
